docs/src/modules/ROOT/pages/using-timefold-solver/modeling-planning-problems.adoc

```diff
@@ -1150,6 +1148,25 @@ public class Job {
 ----
 ====
 
+==== Optimizing Shadow Variables
+
+The Solver analyzes the graph formed `@ShadowVariable` to decide what is the most optimal strategy for updating them.
+For the best performance, follow these guidelines:
+
+- Only declare custom Shadow Variables on a single entity class.
```

Comment:
```suggestion
- Keep all custom Shadow Variables on the same entity class.
If you have a custom Shadow Variable on another entity class that is used to group entities of another class, consider moving the shadow variable to the grouped entity class and use `alignmentKey` to avoid recomputation.
```

The example is somewhat vague and, if I did not already know what this was talking about, I wouldn't have understood it from this description. I wonder if the use case is so common that it deserves a mention; if you keep it, please reference the alignment section, instead of specifically mentioning the `alignmentKey`.